moto/sagemaker/models.py

```diff
@@ -1105,6 +1107,10 @@ def gen_response_object(self) -> Dict[str, Any]:
             "AdditionalInferenceSpecifications",
             "SkipModelValidation",
         ]
+        if self.model_package_type == "Versioned":
```

Comment:
In the first iteration it wasn't obvious what was going on, but now the logic is clear - so this is a much better solution IMO.